### PR TITLE
chore: short dsat e2e tests

### PR DESCRIPTION
**Pull request by garrett361**
## Description

This PR shortens up the `dsat` ([deepspeed autotune](https://docs.determined.ai/latest/model-dev-guide/api-guides/apis-howto/deepspeed/autotuning.html)) e2e tests by having them run only an 4-trial search, rather than the default 64 trials (as controlled by the `--max-trials` flag).

## Test Plan

Inspect the Circle CI test logs and verify that the tests do not come close to hitting the 30 min timeout limit.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by tayritenour — COMMENTED** [review]
So ultimately this would serialized out to use the experiment config with `max_trials: 1`? A couple of questions, wouldn't this prevent the ASHA from doing its job? I thought it needed more trials to iterate properly.

Let's make sure that it does indeed run and that this doesn't undermine the ASHA entirely. Otherwise, looks fine

**Comment by garrett361** [comment]
> So ultimately this would serialized out to use the experiment config with `max_trials: 1`? A couple of questions, wouldn't this prevent the ASHA from doing its job? I thought it needed more trials to iterate properly.

Yeah, it sets `max_trials=1` and is a pretty trivial "search".  The `test_deepspeed_autotuning.py` tests are pretty extensive, so I don't know how much value these e2e tests add, apart from showing that the actual trials do run.  Could also do like `max_trials=4` or `8`? What do you think?

**Comment by garrett361** [comment]
Ah, yeah, I forgot `max_trials` includes the model profiling run. The tests failed, and I think this is why. Trying again with `max_trials=2`

File: harness/determined/pytorch/dsat/__main__.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 def parse_args() -> argparse.Namespace:
     parser = _utils.get_full_parser()
     args = parser.parse_args()
+    assert args.max_trials > 1, "--max-trials must be larger than 1"
```
Comment (garrett361) [inline harness/determined/pytorch/dsat/__main__.py:14]: Added this sanity check because even I forgot about this.

**Comment by tayritenour** [comment]
> Could also do like `max_trials=4` or `8`? What do you think @tayritenour ?

I think that makes sense. Enough to at least make sure it's running in a basic manner (the image isn't broken for some reason, etc)

**Review by tayritenour — APPROVED** [review]
LGTM

**Comment by garrett361** [comment]
Great will bump to `--max-trials 8`, make sure tests succeed quickly enough, then merge

**Review by tayritenour — APPROVED** [review]

**Comment by garrett361** [comment]
Bumped down to 4 trials for more speed.

**Pull request merged**